pkg/backend/httpstate/backend.go

```diff
@@ -828,13 +828,13 @@ func (b *cloudBackend) runEngineAction(
 	<-displayDone
 	scope.Close() // Don't take any cancellations anymore, we're shutting down.
 	close(engineEvents)
-	close(displayEvents)
 	close(displayDone)
 	contract.IgnoreClose(manager)
 
 	// Make sure that the goroutine writing to displayEvents and callerEventsOpt
 	// has exited before proceeding
 	<-eventsDone
+	close(displayEvents)
```

Comment:
i do super hate the mega duplication between these mtehods.  can we not extract anything common here about cleanin ghtis up?

Reply:
I'll file a bug for this.